### PR TITLE
[18.0][MIG] stock_request: Migration to 18.0

### PR DESCRIPTION
**Pull request by ced-adhoc**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@ced-adhoc Thanks for this. Could you check tests?

**Comment by ced-adhoc** [comment]
@rousseldenis Sure! I'm on it! I'm also running into an issue with the view_stock_request_form -it's not displaying correctly at lower zoom levels, and I might need help figuring that part out. I'll let you know when it's done, ASAP.

**Review by mav-adhoc — APPROVED** [review]

**Review by luc-adhoc — APPROVED** [review]
functional review

**Comment by mav-adhoc** [comment]
LGTM!

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Stock request orders view is broken. From runbot:

image

could you have a look?

could you also cherry pick this recent commit https://github.com/OCA/stock-logistics-request/pull/50/commits/fa69ab4441560661ce9b49f839b00cf15eca83ad

Thanks!

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
@ced-adhoc Thanks for attending my comments!

One more detail, the messages posted in the chatter of the stock request are not parsed as html:

image

Could you have a look?

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Thanks for fixing! However it seems that now https://github.com/OCA/stock-logistics-request/commit/fa69ab4441560661ce9b49f839b00cf15eca83ad got removed, could you add it again? Also, CI is red.

**Comment by ced-adhoc** [comment]
Hi @LoisRForgeFlow !  Yes, no worries, I'll cherry-pick your commit once my job is done. The CI error was due to some warnings related to the html message. Please let me know if you have further observations :)

**Comment by LoisRForgeFlow** [comment]
@ced-adhoc Thanks! just let me know when ready to review again, it looks like we are almost there.

File: stock_request/models/stock_move_line.py [hunk 1]
```diff
@@ -10,27 +12,34 @@ class StockMoveLine(models.Model):
     @api.model
     def _stock_request_confirm_done_message_content(self, message_data):
         title = (
-            _("Receipt confirmation %(picking_name)s for your Request %(request_name)s")
+            Markup("<h3>%s<h3>")
+            % _(
+                "Receipt confirmation %(picking_name)s for your Request "
+                "%(request_name)s"
+            )
```
Comment (rousseldenis) [inline stock_request/models/stock_move_line.py]: @ced-adhoc You should use translation function arguments to pass translation value. `_("%(request_name)s", message_data)`

**Comment by ced-adhoc** [comment]
Hi @LoisRForgeFlow , just checking if I need to reorder the commits so your UPD commit comes before my MIG commit. 

@rousseldenis I believe is now aligned with your request.

File: stock_request/models/stock_move_line.py [hunk 2]
```diff
@@ -9,28 +11,34 @@ class StockMoveLine(models.Model):
...
-        )
-        message = "<h3>%s</h3>" % title
-        message += (
-            _(
-                "The following requested items from Stock Request %(request_name)s "
-                "have now been received in %(location_name)s using Picking "
-                "%(picking_name)s:"
-            )
-            % message_data
+        # _("%(request_name)s", message_data)
```
Comment (rousseldenis) [inline stock_request/models/stock_move_line.py]: Please remove this

File: stock_request/models/stock_move_line.py [hunk 3]
```diff
@@ -9,28 +11,34 @@ class StockMoveLine(models.Model):
...
+            "have now been received in %(location_name)s using Picking "
+            "%(picking_name)s:",
+            request_name=message_data["request_name"],
+            location_name=message_data["location_name"],
+            picking_name=message_data["picking_name"],
         )
-        message += "</ul>"
+
+        items = Markup(
+            "<ul><li><b>%(product_name)s</b>: Transferred quantity %(product_qty)s "
```
Comment (rousseldenis) [inline stock_request/models/stock_move_line.py]: Why not using translation here ?

Comment (ced-adhoc) [reply stock_request/models/stock_move_line.py]: Because I missed it by mistake.

**Review by LoisRForgeFlow — APPROVED** [review]
All Good now, thanks :smile:

**Comment by LoisRForgeFlow** [comment]
@rousseldenis good from your side too?

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**